pandas/core/groupby/groupby.py

```diff
@@ -1617,92 +1618,102 @@ def nth(self, n, dropna=None):
         4  2  5.0
         """
 
-        if isinstance(n, int):
-            nth_values = [n]
-        elif isinstance(n, (set, list, tuple)):
-            nth_values = list(set(n))
-            if dropna is not None:
```

Comment:
Instead of explicitly checking `if dropna is not None` this condition was refactored to sit in a branch that follows the implicit condition of `if dropna`. 

There is however a slight behavior difference between this PR and master, where now these are equivalent:

```python
>>> df = pd.DataFrame([[0, 1], [0, 2]], columns=['foo', 'bar'])
>>> df.groupby('foo').nth([0], dropna=None)
     bar
foo
0      1
>>> df.groupby('foo').nth([0], dropna=False)
     bar
foo
0      1
```

Whereas on master these would not yield the same thing:

```python
>>> df = pd.DataFrame([[0, 1], [0, 2]], columns=['foo', 'bar'])
>>> df.groupby('foo').nth([0], dropna=None)
     bar
foo
0      1
>>> df.groupby('foo').nth([0], dropna=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/williamayd/clones/pandas/pandas/core/groupby/groupby.py", line 1626, in nth
    "dropna option with a list of nth values is not supported")
ValueError: dropna option with a list of nth values is not supported
```

I think the new behavior is more consistent and preferred. Might be worth a whatsnew